packages/core/src/StateNode.ts

```diff
@@ -1324,25 +1329,29 @@ class StateNode<
     }
 
     let maybeNextState = nextState;
-    const isTransient = stateNodes.some(stateNode => stateNode._transient);
 
-    if (isTransient) {
-      maybeNextState = this.resolveRaisedTransition(
-        maybeNextState,
-        {
-          type: actionTypes.nullEvent
-        },
-        _event
-      );
-    }
+    if (!isDone) {
+      const isTransient =
```

Comment:
I can't remember right now - how `stateNodes` differ from `configuration`? I know what a configuration is, although I don't remember if we keep the root node in it (SCXML doesn't - but not sure right now if we comply to it in this regard).

Previous version of this condition was not checking `this._transient` at all - so wondering if maybe this has fixed also some other bug on nested state nodes (and not only on the root one).

Reply:
`configuration` does not include the root node in it - it follows SCXML.

Reply:
what about `stateNodes`?

Reply:
We should internally deprecate `stateNodes` for `configuration`, which should be preferred to match the SCXML algorithms.

Reply:
Agreed - that was what i was going for here, wasnt sure how configuration differs from stateNodes so wasnt yet proposing it as I wasnt sure if they are roughly equivalent. Could we then use configuration + root node check for this?